linera-chain/src/manager.rs

```diff
@@ -319,7 +319,11 @@ impl ChainManager {
             }
         }
         let value = HashedCertificateValue::new_timeout(chain_id, height, epoch);
```

Comment:
Why not use a constructor for a `Hashed<Timeout>` directly? (Also below.)

Reply:
b/c the hashes would change, and we still haven't done that.

Reply:
Oh, we are currently instantiating `Hashed<T>`s using `new_unchecked` that actually _don't_ match what `Hashed::new` would do? :grimacing: Would be good to fix that as soon as possible and have only one way to hash things.

Reply:
With this change I get a test failure (without passes fine):
```

failures:
    client::client_tests::test_request_leader_timeout::storage_service
```

Reply:
> Oh, we are currently instantiating `Hashed<T>`s using `new_unchecked` that actually _don't_ match what `Hashed::new` would do? 😬 Would be good to fix that as soon as possible and have only one way to hash things.

The only way to fix this is to get rid of `CertificateValue`.

Reply:
Or to ensure that both kinds are hashed the same way.

Reply:
Exactly: We could serialize `Timeout` as if it were a `CertificateValue::Timeout`.

Reply:
do the transformation in the `Serialize` (and `Deserialize`)? Not a bad idea, I'll give it a try.

Reply:
And maybe not even do a transformation, but serialize a struct that only borrows the payload but is otherwise identical to `CertificateValue`?

Reply:
Here we go again... 
```
     Running tests/format.rs (target/debug/deps/format-3b29b29e5fdad0f7)

running 1 test
test test_format ... FAILED

failures:

---- test_format stdout ----
thread 'test_format' panicked at linera-rpc/tests/format.rs:61:64:
called `Result::unwrap()` on an `Err` value: Custom("Failed to deserialize value: \"Expected a Timeout value\"")
```

Reply:
> And maybe not even do a transformation, but serialize a struct that only borrows the payload but is otherwise identical to `CertificateValue`?

`CertifivateValue` is an enum so my "temporary struct" would have to reflect its structure b/c `serde`(-`reflection`) uses that to derive the binary format.